server_environment/models/server_env_mixin.py

```diff
@@ -0,0 +1,405 @@
+                        'field',
+                        name=is_editable_field,
+                        invisible="1"
+                    )
+                )
+        return view_arch
+
+    def _fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                         submenu=False):
+        view_data = super()._fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu
+        )
+        view_arch = etree.fromstring(view_data['arch'].encode('utf-8'))
+        view_arch = self._server_env_view_set_readonly(view_arch)
+        view_data['arch'] = etree.tostring(view_arch, encoding='unicode')
+        return view_data
+
+    def _server_env_default_fieldname(self, base_field_name):
+        """Return the name of the field with default value"""
+        options = self._server_env_fields[base_field_name]
+        if options and options.get('no_default_field'):
+            return ''
+        return '%s_env_default' % (base_field_name,)
+
+    def _server_env_is_editable_fieldname(self, base_field_name):
+        """Return the name of the field for "is editable"
+
+        This is the field used to tell if the env-computed field can
+        be edited.
+        """
+        return '%s_env_is_editable' % (base_field_name,)
+
+    def _server_env_transform_field_to_read_from_env(self, field):
+        """Transform the original field in a computed field"""
+        field.compute = '_compute_server_env'
+
+        inverse_method_name = '_inverse_server_env_%s' % field.name
+        inverse_method = partialmethod(
+            ServerEnvMixin._inverse_server_env, field.name
+        )
+        setattr(ServerEnvMixin, inverse_method_name, inverse_method)
+        field.inverse = inverse_method_name
+        field.store = False
+        field.required = False
+        field.copy = False
+        field.sparse = None
+        field.prefetch = False
+
+    def _server_env_add_is_editable_field(self, base_field):
+        """Add a field indicating if we can edit the env-computed fields
+
+        It is used in the inverse function of the env-computed field
+        and in the views to add 'readonly' on the fields.
+        """
+        fieldname = self._server_env_is_editable_fieldname(base_field.name)
+        if fieldname not in self._fields:
+            field = fields.Boolean(
+                compute='_compute_server_env_is_editable',
+                automatic=True,
+                # this is required to be able to edit fields
+                # on new records
+                default=True,
+            )
+            self._add_field(fieldname, field)
+
+    def _server_env_add_default_field(self, base_field):
+        """Add a field storing the default value
+
+        The default value is used when there is no key for an env-computed
+        field in the configuration files.
+
+        The field is stored in the serialized field ``server_env_defaults``.
```

Comment:
isn't it stored in db?

Reply:
yes, stored in db in a serialized field, is it contradictory?

Reply:
hmm isn't it store in its own automatic field?
https://github.com/OCA/server-env/pull/13/files#diff-3e560ae060f8d61b271b31acc06ef3b1R271

Reply:
@simahawk  the automatic field is a sparse field, which mean everything ends in ``server_env_defaults`` as json